7.0: make runbot build green

gurneyalex opened the pull request:
- remove type in ir.ui.view xml definitions
- fix a Python SyntaxWarning caused by wrong use of `assert`
- fix the addition of required columns by ensuring they get a value before the NOT NULL sql constraint is added 
- changed WARNING logging to DEBUG where the original logging level was clearly overrated

gurneyalex committed:
remove type in ir.ui.view xml definitions

gurneyalex committed:
fix warning on constraint setting

replace the init method with an overload of _auto_init so that
the lastname column is filled at the time the required constraint is set

pedrobaeza commented:
Still two PEP8 issues:

```
./hr_employee_firstname/hr.py:36:70: E502 the backslash is redundant between brackets
./hr_employee_firstname/hr.py:37:38: F821 undefined name 'pg_varchar'
```

gurneyalex commented:
Hi @pedrobaeza, thanks for the review, but this is still WIP. There are lots of warnings to fix...

gurneyalex committed:
[FIX] fix Python SyntaxWarning

gurneyalex committed:
reduced logging levels

logs at WARNING level should only be used for warnings, not for debugging
information.

gurneyalex committed:
[FIX] hr_leave_accruals: wrong attribute name for _defaults

gurneyalex committed:
[FIX] hr_schedule: provide a default hr.schedule.template

this removes a warning when adding the NO NULL column schedule_template_id
on hr.contract

gurneyalex committed:
[FIX] hr_payroll_period: provide a default schedule

this removes a warning when adding the NO NULL column pps_id
on hr.contract

gurneyalex commented:
Should be fine now. 

I'm not a user of OCA/hr, so I'm not sure what the best option is regarding the two last commits of this PR which create a new default record for a required field of the contracts so that all the contracts have their constraints satisfied when the modules are installed on a pre-existing database.

pedrobaeza commented:
Good job! :+1: 

I also doesn't know exactly that module, but I think the solution you propose is good enough for a start.

gurneyalex commented:
@dufresnedavid having this merged should help getting green builds on runbot for PR targeting the 7.0 branch. Thanks for your review.

hbrunn commented:
:+1:

guewen commented:
:+1:
